python/iceberg/parquet/parquet_reader.py

```diff
@@ -0,0 +1,340 @@
+
+        Returns
+        -------
+        collections.OrderedDict
+            map that maps the file field name to the expected field name
+
+        """
+        return OrderedDict({file_schema.find_field(field.id).name: field.name
+                            for field in expected_schema.as_struct().fields
+                            if file_schema.find_field(field.id) is not None})
+
+    @staticmethod
+    def get_reverse_field_map(file_schema, expected_schema):
+        """
+        Get a map that maps the field name from the expected schema to the file schema field name
+        based on their id's
+
+        Parameters
+        ----------
+        file_schema : iceberg.api.Schema
+            The iceberg schema of the parquet file
+        expected_schema : iceberg.api.Schema
+            The iceberg schema of the expected output
+
+        Returns
+        -------
+        map
+            map that maps the expected field name to the file field name
+        """
+        return {expected_schema.find_field(field.id).name: field.name
+                for field in file_schema.as_struct().fields
+                if expected_schema.find_field(field.id) is not None}
+
+    @staticmethod
+    def migrate_schema(mapping, table):
+        """
+        Re-order the columns in the pyarrow.Table to the expected schema output ordering
+
+        Parameters
+        ----------
+        mapping : collections.OrderedDict
+            An OrderedDict that has the desired ordering and mapping of columns
+        table : pyarrow.Table
+            The pyarrow.Table to be rearranged
+
+        Returns
+        -------
+        pyarrow.Table
+            A pyarrow.Table rearranged to match the naming and ordering is specified by mapping
+
+        """
+        new_columns = []
+        for key, value in mapping.items():
+            column_idx = table.schema.get_field_index(key)
+            column = table[column_idx]
+            new_columns.append(pa.Column.from_array(value, column.data))
+
+        return pa.Table.from_arrays(new_columns)
+
+    @staticmethod
+    def create_null_column(reference_column, name, dtype_tuple):
+        """
+        Create a filled column of Null values
+
+        Parameters
+        ----------
+        reference_column : pyarrow.Column
+            A column to use as the reference for the chunked arrays
+        name : str
+            The column_name
+        dtype_tuple : tuple
+            A tuple that contains the pyarrow datatype and the init value to use for the array
+        Returns
+        -------
+        pyarrow.Table
+            A pyarrow.Table rearranged to match the naming and ordering is specified by mapping
+
+        """
+        dtype, init_val = dtype_tuple
+        chunk = pa.chunked_array([pa.array(np.full(len(c), init_val), type=dtype, mask=[True] * len(c))
+                                  for c in reference_column.data.chunks], type=dtype)
+
+        return pa.Column.from_array(name, chunk)
```

Comment:
Column doesn't exist anymore in `pyarrow==0.15`, just return a tuple of `name, chunk`

Reply:
Noted, i'll amend this